controller/controller_test.go

```diff
@@ -0,0 +1,79 @@
+package controller
+
+import (
+	"testing"
+	"time"
+)
+
+func TestGenerateRandomToken(t *testing.T) {
+	n := 256
+	key, err := GenerateRandomToken(n)
+	if err != nil || len(key) != n {
+		t.Errorf("The signing key is wrong: %s", key)
+	}
+}
+
+func TestGenerateRandomString(t *testing.T) {
+	n := 5
+	s, err := GenerateRandomString(n)
+	if err != nil || len(s) < n {
```

Comment:
why not `len(s) != n` ?

Reply:
@pierrebeaucamp because we are encoding the string to `base64`, and each element of []byte returned by `GenerateRandomToken` is 0-255, we can get elements that represent multiple characters. This can result in strings longer than n.

Reply:
The reason why I'm asking is that the input argument to `GenerateRandomString` is somewhat arbitrary. It's neither the length of the resulting string, nor is it the byte-size of the resulting string. To understand what the `5` in this scenario signals, you need to read through the implementation itself.

Not a huge deal, but I thought it's worth pointing out

Reply:
It is a good point. I have thrown added logic to `GenerateRandomString` that discards the part of the string longer than n characters. Test case updated accordingly